report/verification/mes/tmap_1c.md

```diff
@@ -98,45 +98,31 @@ from scipy.special import erf
 norm = Normalize(vmin=0, vmax=max(profile_times))
 cmap = cm.viridis
 
-plt.figure()
-filename = model.exports[1].filename
-data = np.genfromtxt(filename, delimiter=",", skip_header=1)
-data = np.array(data).T
-
-# sort data by the x-row
-data = np.array(data).T
-sorted_column_ind = data[0].argsort() # indices to sort by x
-data = data[:,sorted_column_ind]
-
-# pre-compute exact solution
-def get_exact_solution(x, t):
+def exact_solution(x, t):
     sqrt_term = np.sqrt(4 * D * t)
     return (
     C_0
     / 2
     * (
-        2 * erf(x / sqrt_term[:, None])
-        - erf((x - preloaded_length) / sqrt_term[:, None])
-        - erf((x + preloaded_length) / sqrt_term[:, None])
+        2 * erf(x / sqrt_term)
+        - erf((x - preloaded_length) / sqrt_term)
+        - erf((x + preloaded_length) / sqrt_term)
     )
 )
 
-t = np.array(profile_times)
-x = data[0]
-exact_solution = get_exact_solution(x, t)
-
+plt.figure()
+filename = model.exports[1].filename
+data = np.genfromtxt(filename, delimiter=",", names=True)
 for i, t in enumerate(profile_times):
-    y = data[i + 1]
-
     label = "exact" if i == 0 else ""
-    plt.plot(
-        x,
-        exact_solution[i],
-        linestyle="dashed",
-        color="tab:grey",
-        linewidth=3,
-        label=label,
-    )
+    x = data["x"]
+    y_name = f"t{t:.2e}s".replace("+", "").replace("-", "").replace(".", "")
+    y = data[y_name]
```

Comment:
Why use names and not indices?

Reply:
It's always better to use names of columns instead of indices. That way, if for some reason the indexing changes (maybe because you added a step in your workflow or another column...) then the code still works.

Otherwise you need to remember to change all the indices and it is overall less robust.

Reply:
Got it, that makes a lot of sense.